gengokit/gengokit.go

```diff
@@ -200,70 +193,23 @@ func templatePathToActual(templFilePath, packageName string) string {
 	return actual
 }
 
-// updateServerMethods accepts NAME-service/handlers/server/server_handlers.go
-// as an io.Reader and modifies it to contains only functions returned by
-// serviceFunctionNames.
-func updateServerMethods(svcHandler io.Reader, te *templateExecutor) (outCode io.Reader, err error) {
-	svcFuncs := serviceFunctionsNames(te.Service.Methods)
-
-	const svcMethodsTemplPath = "NAME-service/partial_template/service.methods"
-	const svcInterfaceTemplPath = "NAME-service/partial_template/service.interface"
-
-	astMod := astmodifier.New(svcHandler)
-
-	astMod.RemoveFunctionsExecpt(svcFuncs)
-	astMod.RemoveInterface("Service")
-
-	// Index the handler functions, apply handler template for all function in
-	// service definition that are not defined in handler
-	currentFuncs := astMod.IndexFunctions()
-	code := astMod.Buffer()
-
-	trimmedTe := te.trimServiceFuncs(currentFuncs)
-
-	newFuncs, err := applyTemplateFromPath(svcMethodsTemplPath, trimmedTe)
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to apply service methods template")
-	}
-
-	code.ReadFrom(newFuncs)
-
-	// Insert updated Service interface
-	svcInterface, err := applyTemplateFromPath(svcInterfaceTemplPath, te)
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to apply service interface template")
-	}
-
-	code.ReadFrom(svcInterface)
-
-	return code, nil
-}
-
-// updateClientMethods accepts NAME-service/handlers/client/client_handler.go
-// as an io.Reader and modifies it to contains only functions returned by
-// serviceFunctionNames.
-func updateClientMethods(clientHandler io.Reader, te *templateExecutor) (outCode io.Reader, err error) {
+func updateMethods(handler io.Reader, templ []byte, te *templateExecutor) (io.Reader, error) {
```

Comment:
The function is going to be removed in the next few days into another package, I feel the intent of the function is clear enough in the code, and it is unexported so it will not create doc comments.